SDBlockDevice.cpp

```diff
@@ -558,6 +558,17 @@ int SDBlockDevice::erase(bd_addr_t addr, bd_size_t size)
     }
     int status = BD_ERROR_OK;
 
+    // Align erases, missed blocks will be left unerased, but this is fine as
+    // the block device api doesn't garuntee the final state of erased bytes and
+    // a following program will still work fine.
+    bd_size_t misalignment = _erase_size - (addr % _erase_size);
```

Comment:
Looks good, I'd go for something simpler such as
```c
if (size < _erase_size)
    return status; // it's probably wise to inform the user the operation is unsupported
```

as the controller already rounds the start/end addresses according to the required granularity.